lib/client.js

```diff
@@ -27,9 +27,52 @@ class Client {
     this.queue = Promise.resolve();
     this.totalUploaded = 0;
     this.version = JSON.parse(fs.readFileSync(join(__dirname, '..', 'package.json')).toString()).version;
+    this.executionList = Promise.resolve();
+
     console.log(APP_PREFIX, `Testomatio Reporter v${this.version}`);
   }
 
+  /**
+   * Asynchronously prepares the execution list for running tests through various pipes.
+   * Each pipe in the client is checked for enablement, and if all pipes are disabled, the function returns a resolved Promise.
+   * Otherwise, it executes the `prepareRun` method for each enabled pipe and collects the results.
+   * The results are then filtered to remove any undefined values.
+   * If no valid results are found, the function returns undefined.
+   * Otherwise, it returns the first non-empty array from the filtered results.
+   *
+   * @param {Object} opts - The options for preparing the test execution list.
+   * @returns {Promise<any>} - A Promise that resolves to an array containing the prepared execution list,
+   * or resolves to undefined if no valid results are found or if all pipes are disabled.
+   */
+  async prepareRun(opts) {
+    // all pipes disabled, skipping
+    if (!this.pipes.some(p => p.isEnabled)) {
+      return Promise.resolve();
+    }
+
+    try {
+      const results = await Promise.all(this.pipes.map(async p => {
+        try {
+          return await p.prepareRun(opts);
+        } catch (err) {
+          console.log(APP_PREFIX, p.toString(), err);
+          return;
+        }
+      }));
+  
+      const filteredResults = results.filter(result => result !== undefined);
```

Comment:
looks like a very complex logic
I would directly set the pipe name inside the filter 

```
npx start-test-run --filter 'testomatio:tag=xxx'

npx start-test-run --filter 'testomatio:label=xxx'
```

Instead of guessing which pipe to pass data, we will find Testomatio pipe but name and we will pass the `label=xxx` into it.

Reply:
It seems to me that it will be very difficult for the end user to understand such a form of command, but I have implemented your approach! If so, you can remove this processing line.